core/src/test/java/org/apache/spark/memory/TaskMemoryManagerSuite.java

```diff
@@ -25,27 +26,37 @@
 
 public class TaskMemoryManagerSuite {
 
+  static TaskMemoryManager manager;
+
+  @After
+  public void after() {
+    Assert.assertEquals(0, manager.getMemoryConsumptionForThisTask());
```

Comment:
@squito had the same concern. I purposefully leaked some memory at the end of leakedPageMemoryIsDetected and got this very helpful trace:

[info] Test org.apache.spark.memory.TaskMemoryManagerSuite.leakedPageMemoryIsDetected started
[error] Test org.apache.spark.memory.TaskMemoryManagerSuite.leakedPageMemoryIsDetected failed: java.lang.AssertionError: expected:<0> but was:<4096>, took 0.007 sec
[error]     at org.apache.spark.memory.TaskMemoryManagerSuite.after(TaskMemoryManagerSuite.java:34)
[error]     ...